src/Components/useField.ts

```diff
@@ -0,0 +1,21 @@
+import { ChangeEventHandler, useCallback, useState } from 'react';
+
+type UseFieldType<V = undefined, Element = HTMLInputElement> = [V, ChangeEventHandler<Element>];
+
+export const useField = <
```

Comment:
Expertly written hook! Beautiful. It's like an expert made it...

Reply:
Thanks, sometimes I just managed to pull one out of the bag.